tests/v2/validation/provisioning/k3s/custom_cluster_test.go

```diff
@@ -221,13 +221,24 @@ func (c *CustomClusterProvisioningTestSuite) testProvisioningK3SCustomCluster(cl
 	err = wait.WatchWait(result, checkFunc)
 	assert.NoError(c.T(), err)
 	assert.Equal(c.T(), clusterName, clusterResp.ObjectMeta.Name)
+	assert.Equal(c.T(), kubeVersion, cluster.Spec.KubernetesVersion)
+
+	clusterIDName, err := clusters.GetClusterIDByName(c.client, clusterName)
+	assert.NoError(c.T(), err)
+
+	err = nodestat.IsNodeReady(client, clusterIDName)
+	require.NoError(c.T(), err)
 
 	clusterToken, err := clusters.CheckServiceAccountTokenSecret(client, clusterName)
 	require.NoError(c.T(), err)
 	assert.NotEmpty(c.T(), clusterToken)
 
-	if harden.Hardened && kubeVersion < "v1.25.0" {
-		err = hardening.HardeningNodes(client, harden.Hardened, nodes, nodesAndRoles)
+	podResults, podErrors := pods.StatusPods(client, clusterIDName)
+	assert.NotEmpty(c.T(), podResults)
+	assert.Empty(c.T(), podErrors)
+
+	if c.hardened && kubeVersion < string(provisioning.HardenedKubeVersion) {
```

Comment:
if the last comment uses the `1.x.99` default, this technically would need to change to <=, though I doubt we will ever hit that patch version for any k8s release...

Reply:
Fixed.